src/rnnlm/rnnlm-example-utils.h

```diff
@@ -33,6 +33,8 @@
 namespace kaldi {
 namespace rnnlm {
 
+const int64 MAX_MAT_SIZE = 1000000000;
```

Comment:
It's rather ugly to make this a const int like this (and it would, I think, be misnamed, should probably be kMaxMatSize).
Put it in the config class.  And clarify the units, whether it's bytes or elements.  e.g.  max-logprob-elements or max-logprob-bytes.